### PR TITLE
Fix default template for content type which contains block with schem…

### LINKED ISSUE #5158: Adding default template for content type with block(having schemaEnhancer) throwing error.
If you have a schema enhancer for a particular block and then you try to add this block in the layout options of content type will throw error.

Steps to reproduce the error.

1. Create a schemaEnhancer for a particular block
2. Go to any content type and dexterity content layout and select this block.

You will get error.

### PR DESCRIPTION
**Pull request by iFlameing**
…aEnhancer.

Fixes #5158

### CODE REVIEW

**Review by davisagli — CHANGES_REQUESTED** [review]
@iFlameing This doesn't look like the right fix to me. We need to figure out why the schema enhancer breaks and fix it, not disable the enhancers.

**Comment by davisagli** [comment]
@iFlameing Oh, now I understand. You are turning off schema extenders only for the block settings schema. Let me check whether this is the best place to do it, but that makes more sense.

**Review by davisagli — COMMENTED** [review]
@iFlameing I refactored so that the logic for whether to apply the enhancers is in BlockDataForm, not withVariationSchemaEnhancer. The job of withVariationSchemaEnhancer is to actually apply the enhancers, not decide whether it's appropriate to do that.

File: news/5158.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix default template for content type which contains block with schemaEnhancer. @iFlameing
```
Comment (davisagli) [inline news/5158.bugfix:1]:
```suggestion
Fix editing layout for blocks using schema enhancers. @iFlameing, @davisagli
```

**Review by davisagli — APPROVED** [review]
@iFlameing I also added tests. I think it's ready now.

**Review by avoinea — APPROVED** [review]

**Review by sneridagh — APPROVED** [review]

**Pull request merged**